### PR TITLE
Resovler for finding Unlinked ToolSed installed dependencies 

### PR DESCRIPTION
**Pull request by Christian-B**
Overview:
This provides a plugin Resolver that will find dependencies previously installed via the toolshed.
Mainly intended for testing of either manually installed tools or using planemo
This resolver will look for the env.sh or bin directory not just in dependency_dir/package/version but also in dependency_dir/package/version/owner/package_name/revision.

Implementation
As there can be more than one revision for a particular dependency/version this resolver has a two staged approach.
1. Find all candidate dependencies (include at the normal manual locations), logging dependencies found
2. Select the preferred one.
   By default first by preferred owner, then by modified time
   Logging method used to pick "preferred" revision

Installation:
Requires a dependency_resolvers_config.xml (see documentation)

Testing:
Tested by making a copy of a working tool with requirements and addint it manually to toot confi
Tested with a Toot shed installed tool using planemo
Both failed without this and passed with

For example I tested on:
https://toolshed.g2.bx.psu.edu/repository?repository_id=2df7e24ce6c1f224

### CODE REVIEW

**Comment by Christian-B** [comment]
With thanks to @nsoranzo  and @jmchilton for their helpful comments made in https://github.com/galaxyproject/galaxy/pull/413

File: lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+Ideally all dependencies will be stored in the database 
+    when a tool is added from a toolshed.
+That should remain the preferred way of locating dependencies.
+
+In cases where that is not possible 
+    for example during testing this resolver can act as a backup.
+This resolver looks not just for manually added dependencies 
+    but also ones added from a toolshed.
+
+This tool is still under development so the default behaviour could change.
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py:14]: s/tool/resolver/ ?

File: lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+It has been tested when placed in the same directory as galaxy_packages.py
+
+At the time of writting July 3 2015 this resolver has to be pluged in.
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py:17]: s/pluged/plugged/

File: lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+See bottom for instuctions on how to add this resolver.
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/resolvers/unlinked_tool_shed_packages.py:18]: s/instuctions/instructions/

**Comment by nsoranzo** [comment]
Hi @Christian-B, thanks for the updated contribution!
Can you please fix the flake8 errors described at https://travis-ci.org/galaxyproject/galaxy/jobs/69455002 ?

Thanks!

**Comment by Christian-B** [comment]
Thank you for the useful comments.

I realised that I have done this against the wrong (my dev) branch.

I am closing this and will reopen a new improved pull request against a specific branch,

This will of course include the fixes suggested above.

**Comment by Christian-B** [comment]
This as been resubmitted under a separate branch.

See https://github.com/galaxyproject/galaxy/pull/441

With the above spelling errors (and many more) fixed.

**Pull request closed without merging**